icechunk-python/python/icechunk/__init__.py

```diff
@@ -626,35 +603,40 @@ def supports_listing(self) -> bool:
     def supports_deletes(self) -> bool:
         return self._store.supports_deletes
 
-    def list(self) -> AsyncGenerator[str, None]:
```

Comment:
The change from async generator to async iterator doesnt change anything on the rust side, its the same prototype